File: Esercizio Esame (Customer)/src/Entity/Order.php
<?php
    
    namespace Entity;
    
    class Order
    {
        
        private $id;
        private $transaction;
        private $status;
        private $owner_id;
        private $created_at;
        private $shipment_type;
        private $payment_method_id;
        private $shipping_address_id;
        private $code;
        
        private $products;
        
        private $item_count;
        
        private $total;
        private $delivered_at;
        private $tracking_code;
    
        /**
         * @return mixed
         */
        public function getId()
        {
            return $this->id;
        }
    
        /**
         * @param mixed $id
         */
        public function setId($id): void
        {
            $this->id = $id;
        }
    
        /**
         * @return mixed
         */
        public function getTransaction()
        {
            return $this->transaction;
        }
    
        /**
         * @param mixed $transaction
         */
        public function setTransaction($transaction): void
        {
            $this->transaction = $transaction;
        }
    
        /**
         * @return mixed
         */
        public function getStatus()
        {
            return $this->status;
        }
    
        /**
         * @param mixed $status
         */
        public function setStatus($status): void
        {
            $this->status = $status;
        }
    
        /**
         * @return mixed
         */
        public function getOwnerId()
        {
            return $this->owner_id;
        }
    
        /**
         * @param mixed $owner_id
         */
        public function setOwnerId($owner_id): void
        {
            $this->owner_id = $owner_id;
        }
    
        /**
         * @return mixed
         */
        public function getCreatedAt()
        {
            return $this->created_at;
        }
    
        /**
         * @param mixed $created_at
         */
        public function setCreatedAt($created_at): void
        {
            $this->created_at = $created_at;
        }
    
        /**
         * @return mixed
         */
        public function getShipmentType()
        {
            return $this->shipment_type;
        }
    
        /**
         * @param mixed $shipment_type
         */
        public function setShipmentType($shipment_type): void
        {
            $this->shipment_type = $shipment_type;
        }
    
        /**
         * @return mixed
         */
        public function getPaymentMethodId()
        {
            return $this->payment_method_id;
        }
    
        /**
         * @param mixed $payment_method_id
         */
        public function setPaymentMethodId($payment_method_id): void
        {
            $this->payment_method_id = $payment_method_id;
        }
    
        /**
         * @return mixed
         */
        public function getShippingAddressId()
        {
            return $this->shipping_address_id;
        }
    
        /**
         * @param mixed $shipping_address_id
         */
        public function setShippingAddressId($shipping_address_id): void
        {
            $this->shipping_address_id = $shipping_address_id;
        }
    
        /**
         * @return mixed
         */
        public function getProducts()
        {
            return $this->products;
        }
    
        /**
         * @param mixed $products
         */
        public function setProducts($products): void
        {
            $this->products = $products;
        }
    
        /**
         * @return mixed
         */
        public function getCode()
        {
            return $this->code;
        }
    
        /**
         * @param mixed $code
         */
        public function setCode($code): void
        {
            $this->code = $code;
        }
    
        /**
         * @return mixed
         */
        public function getItemCount()
        {
            return $this->item_count;
        }
    
        /**
         * @param mixed $item_count
         */
        public function setItemCount($item_count): void
        {
            $this->item_count = $item_count;
        }
    
        /**
         * @return mixed
         */
        public function getTotal()
        {
            return $this->total;
        }
    
        /**
         * @param mixed $total
         */
        public function setTotal($total): void
        {
            $this->total = $total;
        }
        
        public function getStatusString(){
            switch ($this->status){
                case 0:
                    return '<span class="text-warning">Under processing</span>';
                case 1:
                    return '<span class="text-primary">Sent (Tracking code: '.$this->tracking_code.')</span>';
                case 2:
                    return '<span class="text-success">Delivered '.$this->delivered_at.'</span>';
                    
            }
            return '<span class="text-danger">Error</span>';
        }
    
        /**
         * @return mixed
         */
        public function getDeliveredAt()
        {
            return $this->delivered_at;
        }
    
        /**
         * @param mixed $delivered_at
         */
        public function setDeliveredAt($delivered_at): void
        {
            $this->delivered_at = $delivered_at;
        }
    
        /**
         * @return mixed
         */
        public function getTrackingCode()
        {
            return $this->tracking_code;
        }
    
        /**
         * @param mixed $tracking_code
         */
        public function setTrackingCode($tracking_code): void
        {
            $this->tracking_code = $tracking_code;
        }
    
    }

File: Esercizio Esame (Admin)/src/upo/gaborgalazzo/mweb/marketplace/domain/Order.java
package upo.gaborgalazzo.mweb.marketplace.domain;

import upo.gaborgalazzo.mweb.marketplace.functiolanities.StringConverter;

import java.sql.Date;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class Order
{
	private int id;
	private String transaction;
	private int status;
	private int ownerId;
	private Date createdAt;
	private int shipmentType;
	private int paymentMethodId;
	private int shippingAddressId;
	private String code;
	private int itemCount;
	private float total;
	private Date deliveredAt;
	private String trackingCode;
	private String ownerInfo;
	private String pmInfo;
	private String saInfo;

	private List<CartProduct> products = new ArrayList<>();
	private int messagesCount;

	public static Order fromResultSet(ResultSet resultSet) throws SQLException
	{
		Order order = new Order();
		order.setId(resultSet.getInt("id"));
		order.setStatus(resultSet.getInt("status"));
		order.setOwnerId(resultSet.getInt("owner_id"));
		order.setShipmentType(resultSet.getInt("shipment_type"));
		order.setPaymentMethodId(resultSet.getInt("payment_method_id"));
		order.setShippingAddressId(resultSet.getInt("shipping_address_id"));
		order.setItemCount(resultSet.getInt("item_count"));
		order.setMessagesCount(resultSet.getInt("messages_count"));
		order.setTransaction(StringConverter.toUtf8(resultSet,"transaction"));
		order.setCode(StringConverter.toUtf8(resultSet,"code"));
		order.setTrackingCode(StringConverter.toUtf8(resultSet,"tracking_code"));
		order.setCreatedAt(resultSet.getDate("created_at"));
		order.setDeliveredAt(resultSet.getDate("delivered_at"));
		order.setTotal(resultSet.getFloat("total"));
		order.setOwnerInfo(StringConverter.toUtf8(resultSet,"owner_info"));
		order.setPmInfo(StringConverter.toUtf8(resultSet,"pm_info"));
		order.setSaInfo(StringConverter.toUtf8(resultSet,"sa_info"));
		return order;
	}

	public int getId()
	{
		return id;
	}

	public void setId(int id)
	{
		this.id = id;
	}

	public String getTransaction()
	{
		return transaction;
	}

	public void setTransaction(String transaction)
	{
		this.transaction = transaction;
	}

	public int getStatus()
	{
		return status;
	}

	public void setStatus(int status)
	{
		this.status = status;
	}

	public int getOwnerId()
	{
		return ownerId;
	}

	public void setOwnerId(int ownerId)
	{
		this.ownerId = ownerId;
	}

	public Date getCreatedAt()
	{
		return createdAt;
	}

	public void setCreatedAt(Date createdAt)
	{
		this.createdAt = createdAt;
	}

	public int getShipmentType()
	{
		return shipmentType;
	}

	public void setShipmentType(int shipmentType)
	{
		this.shipmentType = shipmentType;
	}

	public int getPaymentMethodId()
	{
		return paymentMethodId;
	}

	public void setPaymentMethodId(int paymentMethodId)
	{
		this.paymentMethodId = paymentMethodId;
	}

	public int getShippingAddressId()
	{
		return shippingAddressId;
	}

	public void setShippingAddressId(int shippingAddressId)
	{
		this.shippingAddressId = shippingAddressId;
	}

	public String getCode()
	{
		return code;
	}

	public void setCode(String code)
	{
		this.code = code;
	}

	public int getItemCount()
	{
		return itemCount;
	}

	public void setItemCount(int itemCount)
	{
		this.itemCount = itemCount;
	}

	public float getTotal()
	{
		return total;
	}

	public void setTotal(float total)
	{
		this.total = total;
	}

	public Date getDeliveredAt()
	{
		return deliveredAt;
	}

	public void setDeliveredAt(Date deliveredAt)
	{
		this.deliveredAt = deliveredAt;
	}

	public String getTrackingCode()
	{
		return trackingCode;
	}

	public void setTrackingCode(String trackingCode)
	{
		this.trackingCode = trackingCode;
	}

	public String getOwnerInfo()
	{
		return ownerInfo;
	}

	public void setOwnerInfo(String ownerInfo)
	{
		this.ownerInfo = ownerInfo;
	}

	public String getPmInfo()
	{
		return pmInfo;
	}

	public void setPmInfo(String pmInfo)
	{
		this.pmInfo = pmInfo;
	}

	public String getSaInfo()
	{
		return saInfo;
	}

	public void setSaInfo(String saInfo)
	{
		this.saInfo = saInfo;
	}

	public String getStatusString(){
		switch (status){
			case 0:
				return "<span class='text-warning'>To process</span>";
			case 1:
				return "<span class='text-primary'>Sent</span>";
			case 2:
				return "<span class='text-success'>Delivered</span>";

		}
		return "<span class='text-danger'>Error</span>";
	}

	public List<CartProduct> getProducts()
	{
		return products;
	}

	public void setProducts(List<CartProduct> products)
	{
		this.products = products;
	}

	public void setMessagesCount(int messagesCount)
	{
		this.messagesCount = messagesCount;
	}

	public int getMessagesCount()
	{
		return messagesCount;
	}
}
